tests/phpunit/tests/kses.php

```diff
@@ -1430,6 +1419,79 @@ public static function data_possible_custom_data_attributes_and_transformed_name
 		);
 	}
 
+	/**
+	 * Ensures that only allowable custom data attributes are retained.
+	 *
+	 * @ticket 33121
+	 *
+	 * @dataProvider data_data_attributes_and_whether_they_are_allowed
+	 *
+	 * @param string $attribute_name Custom data attribute, e.g. "data-wp-bind--enabled".
+	 * @param bool   $is_allowed     Whether the given attribute should be allowed.
+	 */
+	public function test_wp_kses_attr_data_attribute_is_allowed( $attribute_name, $is_allowed ) {
+		$element = "<div {$attribute_name}>Pens and pencils.</div>";
```

Comment:
Can do! Out of curiosity, why do this, since `kses` doesn't make any distinction for whether to allow a data attribute or not based on its value?

Reply:
> Out of curiosity, why do this, since `kses` doesn't make any distinction for whether to allow a data attribute or not based on its value?

Mainly as a defensive strategy in case a future maintainer (quite possibly us) makes a mistake. The execution path of KSES is ~a little~ _very_ confusing.

Reply:
added a second test for value-containing attributes in 0498bada52